test/unit/spec/LoginRouter_spec.js

```diff
@@ -189,29 +189,77 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, Logger, OktaAuth, Util, Exp
       expect(fn).not.toThrow();
       expectUnexpectedFieldLog('foo');
     });
-    it('throws a ConfigError if el is not passed as a widget param', function () {
+    it('logs a ConfigError error if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrow();
+      expect(DevLogger.warn).toHaveBeenCalled();
     });
     it('has the correct error message if el is not passed as a widget param', function () {
       var fn = function () { setup({ el: undefined }); };
-      expect(fn).toThrowError('"el" is a required widget parameter');
+      expect(fn).not.toThrow();
+      var err = DevLogger.warn.calls.mostRecent().args[0];
+      expect(err.name).toBe('CONFIG_ERROR');
+      expect(err.message).toEqual('"el" is a required widget parameter');
     });
-    it('throws a ConfigError if baseUrl is not passed as a widget param', function () {
+    it('logs a ConfigError if baseUrl is not passed as a widget param', function () {
       var fn = function () { setup({ baseUrl: undefined }); };
-      expect(fn).toThrowError(Errors.ConfigError);
+      expect(fn).not.toThrowError(Errors.ConfigError);
+      expect(DevLogger.warn).toHaveBeenCalled();
     });
     it('has the correct error message if baseUrl is not passed as a widget param', function () {
       var fn = function () { setup({ baseUrl: undefined }); };
-      expect(fn).toThrowError('"baseUrl" is a required widget parameter');
+      expect(fn).not.toThrow();
+      var err = DevLogger.warn.calls.mostRecent().args[0];
+      expect(err.name).toBe('CONFIG_ERROR');
+      expect(err.message).toEqual('"baseUrl" is a required widget parameter');
+    });
+    itp('uses a default globalSuccessFn if an undefined globalSuccessFn is passed as a widget param', function () {
+      return setup({ globalSuccessFn: undefined })
+      .then(function (test) {
+        Util.mockRouterNavigate(test.router);
+        test.router.navigate('signin/recovery-question');
+        return Expect.waitForPrimaryAuth();
+      })
+      .then(function () {
+        var form = new PrimaryAuthForm($sandbox);
+        expect(form.isPrimaryAuth()).toBe(true);
+      });
+    });
+    itp('uses a default globalSuccessFn if a null globalSuccessFn is passed as a widget param', function () {
+      return setup({ globalSuccessFn: null })
+      .then(function (test) {
```

Comment:
Super nit: you can just move these into a method on its own and just pass `null` and `undefined` as parameters. Or even pass in the `{ globalSuccessFn: null }` or `{ globalErrorFn: undefined }` config.

Reply:
Cleaned this up quite a bit!